src/main/java/com/google/api/gax/grpc/ServiceApiSettings.java

```diff
@@ -200,14 +202,42 @@ public Builder provideChannelWith(final ManagedChannel channel, final boolean sh
     }
 
     /**
-     * Provides the connection settings necessary to create a channel.
+     * Provides the connection settings necessary to create a channel. This will overwrite any
+     * previous calls to provideChannelWith, including calls providing Credentials or a
+     * ManagedChannel.
      */
-    public Builder provideChannelWith(
-        final ConnectionSettings settings) {
+    public Builder provideChannelWith(final ConnectionSettings settings) {
       channelProvider = createChannelProvider(settings);
       return this;
     }
 
+    protected abstract ConnectionSettings getDefaultConnectionSettings();
```

Comment:
default immutable objects should be on the immutable class, as opposed to on the builder.